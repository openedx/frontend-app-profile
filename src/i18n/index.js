import { messages as headerMessages } from '@edx/frontend-component-header';
import { messages as footerMessages } from '@edx/frontend-component-footer';
import { messages as paragonMessages } from '@openedx/paragon';
import arMessages from './messages/ar.json';
import deMessages from './messages/de.json';
import dedeCAMessages from './messages/de_DE.json';
import es419Messages from './messages/es_419.json';
import faIRMessages from './messages/fa_IR.json';
import frCAMessages from './messages/fr_CA.json';
import itMessages from './messages/it.json';
import ititCAMessages from './messages/it_IT.json';
import frMessages from './messages/fr.json';
import hiMessages from './messages/hi.json';
import ptMessages from './messages/pt.json';
import ptptCAMessages from './messages/pt_PT.json';
import ruMessages from './messages/ru.json';
import ukMessages from './messages/uk.json';
import zhcnMessages from './messages/zh_CN.json';
// no need to import en messages-- they are in the defaultMessage field

const appMessages = {
  ar: arMessages,
  'es-419': es419Messages,
  'fa-ir': faIRMessages,
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': dedeCAMessages,
  'it-it': ititCAMessages,
  'pt-pt': ptptCAMessages,
};

export default [
  headerMessages,
  footerMessages,
  paragonMessages,
  appMessages,
];
