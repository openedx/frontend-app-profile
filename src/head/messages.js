import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  'profile.page.title': {
    id: 'profile.page.title',
    defaultMessage: 'Profile | {siteName}',
    description: 'Title tag',
  },
});

export default messages;
