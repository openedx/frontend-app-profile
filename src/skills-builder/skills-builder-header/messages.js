import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  skillsBuilderHeaderTitle: {
    id: 'skills.builder.header.title',
    defaultMessage: 'Skills Builder',
    description: 'Title for the Skills Builder feature',
  },
  skillsBuilderHeaderSubheading: {
    id: 'skills.builder.header.subheading',
    defaultMessage: 'Let edX be your guide',
    description: 'Subheading to the Skills Builder title in the header component',
  },
});

export default messages;
