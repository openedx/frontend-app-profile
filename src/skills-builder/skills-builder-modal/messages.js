import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  /* Modal Action Row Buttons */
  goBackButton: {
    id: 'go.back.button',
    defaultMessage: 'Go Back',
    description: 'Button that sends the user to the previous step in the skills builder.',
  },
  nextStepButton: {
    id: 'next.step.button',
    defaultMessage: 'Next Step',
    description: 'Button that sends the user to the next step in the skills builder.',
  },
  exitButton: {
    id: 'exit.button',
    defaultMessage: 'Exit',
    description: 'Button that exits the Skills Builder.',
  },
});

export default messages;
