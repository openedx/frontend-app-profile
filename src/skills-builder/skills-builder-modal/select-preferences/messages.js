import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  skillsBuilderDescription: {
    id: 'skills.builder.description',
    defaultMessage: 'Find the right courses and programs that help you reach your goals.',
    description: 'Description of what the Skills Builder seeks to accomplish',
  },
  learningGoalPrompt: {
    id: 'learning.goal.prompt',
    defaultMessage: 'First, tell us what you want to achieve',
    description: 'Prompts the user to select their current goal.',
  },
  selectLearningGoal: {
    id: 'select.learning.goal',
    defaultMessage: 'Select a goal',
    description: 'Placeholder text for the goal selection component.',
  },
  learningGoalStartCareer: {
    id: 'learning.goal.start_career',
    defaultMessage: 'I want to start my career',
    description: 'Selected by user if their goal is to start their career.',
  },
  learningGoalAdvanceCareer: {
    id: 'learning.goal.advance_career',
    defaultMessage: 'I want to advance my career',
    description: 'Selected by user if their goal is to advance their career.',
  },
  learningGoalChangeCareer: {
    id: 'learning.goal.change_career',
    defaultMessage: 'I want to change careers',
    description: 'Selected by user if their goal is to change careers.',
  },
  learningGoalSomethingNew: {
    id: 'learning.goal.something.new',
    defaultMessage: 'I want to learn something new',
    description: 'Selected by user if their goal is to learn something new.',
  },
  learningGoalSomethingElse: {
    id: 'learning.goal.something.else',
    defaultMessage: 'Something else',
    description: 'Selected by user if their goal is not described by the other choices.',
  },
  jobTitlePrompt: {
    id: 'job.title.prompt',
    defaultMessage: 'Next, search and select your current job title',
    description: 'Prompts the user to select their current job title or occupation.',
  },
  studentCheckboxPrompt: {
    id: 'student.checkbox.prompt',
    defaultMessage: 'I\'m a student',
    description: 'Label text for the corresponding checkbox',
  },
  currentlyLookingCheckboxPrompt: {
    id: 'currently.looking.checkbox.prompt',
    defaultMessage: 'I\'m currently looking for work',
    description: 'Label text for the corresponding checkbox',
  },
  careerInterestPrompt: {
    id: 'career.interest.prompt',
    defaultMessage: 'What careers are you interested in?',
    description: 'Prompts the user to select careers they are interested in pursuing.',
  },
  removeCareerInterestButtonAltText: {
    id: 'career.interest.remove.button.alt.text',
    defaultMessage: 'Remove career interest: ',
  },
});

export default messages;
