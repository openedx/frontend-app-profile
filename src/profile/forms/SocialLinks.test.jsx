import { mount } from 'enzyme';
import PropTypes from 'prop-types';
import React, { useMemo } from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { configure as configureI18n, IntlProvider } from '@edx/frontend-platform/i18n';
import { getConfig } from '@edx/frontend-platform';
import { AppContext } from '@edx/frontend-platform/react';

import SocialLinks from './SocialLinks';
import * as savingEditedBio from '../__mocks__/savingEditedBio.mockStore';
import messages from '../../i18n';

const mockStore = configureMockStore([thunk]);

const defaultProps = {
  formId: 'socialLinks',
  socialLinks: [
    {
      platform: 'facebook',
      socialLink: 'https://www.facebook.com/aloha',
    },
    {
      platform: 'twitter',
      socialLink: 'https://www.twitter.com/ALOHA',
    },
  ],
  drafts: {},
  visibilitySocialLinks: 'private',
  editMode: 'static',
  saveState: null,
  error: null,
  changeHandler: jest.fn(),
  submitHandler: jest.fn(),
  closeHandler: jest.fn(),
  openHandler: jest.fn(),
};

configureI18n({
  loggingService: { logError: jest.fn() },
  config: {
    ENVIRONMENT: 'production',
    LANGUAGE_PREFERENCE_COOKIE_NAME: 'yum',
  },
  messages,
});

const SocialLinksWrapper = (props) => {
  const contextValue = useMemo(() => ({
    authenticatedUser: { userId: null, username: null, administrator: false },
    config: getConfig(),
  }), []);
  return (
    <AppContext.Provider
      value={contextValue}
    >
      <IntlProvider locale="en">
        <Provider store={props.store}>
          <SocialLinks {...props} />
        </Provider>
      </IntlProvider>
    </AppContext.Provider>
  );
};

SocialLinksWrapper.defaultProps = {
  store: mockStore(savingEditedBio),
};

SocialLinksWrapper.propTypes = {
  store: PropTypes.shape({}),
};

const SocialLinksWrapperWithStore = ({ store }) => {
  const contextValue = useMemo(() => ({
    authenticatedUser: { userId: null, username: null, administrator: false },
    config: getConfig(),
  }), []);
  return (
    <AppContext.Provider
      value={contextValue}
    >
      <IntlProvider locale="en">
        <Provider store={mockStore(store)}>
          <SocialLinks {...defaultProps} formId="bio" />
        </Provider>
      </IntlProvider>
    </AppContext.Provider>
  );
};

SocialLinksWrapperWithStore.defaultProps = {
  store: mockStore(savingEditedBio),
};

SocialLinksWrapperWithStore.propTypes = {
  store: PropTypes.shape({}),
};

describe('<SocialLinks />', () => {
  ['certificates', 'bio', 'goals', 'socialLinks'].forEach(editMode => (
    it(`calls social links with edit mode ${editMode}`, () => {
      const component = <SocialLinksWrapper {...defaultProps} formId={editMode} />;
      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    })
  ));

  it('calls social links with editing', () => {
    const changeHandler = jest.fn();
    const submitHandler = jest.fn();
    const closeHandler = jest.fn();
    const component = (
      <SocialLinksWrapper
        {...defaultProps}
        formId="bio"
        changeHandler={changeHandler}
        submitHandler={submitHandler}
        closeHandler={closeHandler}
      />
    );
    const wrapper = mount(component);
    const socialLink = wrapper.find(SocialLinks);
    const { platform } = defaultProps.socialLinks[0];
    const inputField = socialLink.find(`#social-${platform}`);
    inputField.simulate('change', { target: { value: 'test', name: platform } });
    expect(changeHandler).toHaveBeenCalledTimes(1);

    expect(socialLink.find('#visibilitySocialLinks select').props().value).toBe('private');
    const event = { target: { value: 'all_users', name: 'visibilitySocialLinks' } };
    socialLink.find('#visibilitySocialLinks select').simulate('change', event);
    expect(changeHandler).toHaveBeenCalledTimes(2);

    socialLink.find('[aria-labelledby="editing-form"]').simulate('submit');
    expect(submitHandler).toHaveBeenCalledTimes(1);

    socialLink.find('[aria-labelledby="editing-form"]').find('Button .btn-link').simulate('click');
    expect(closeHandler).toHaveBeenCalledTimes(1);
  });

  it('calls social links with static', () => {
    const openHandler = jest.fn();
    const component = (
      <SocialLinksWrapper
        {...defaultProps}
        formId="goals"
        openHandler={openHandler}
      />
    );
    const wrapper = mount(component);
    const socialLink = wrapper.find(SocialLinks);

    socialLink.find('EmptyContent button').first().simulate('click');
    expect(openHandler).toHaveBeenCalledTimes(1);
  });

  it('calls social links with error', () => {
    const newStore = JSON.parse(JSON.stringify(savingEditedBio));
    newStore.profilePage.errors.bio = { userMessage: 'error' };

    const component = <SocialLinksWrapperWithStore store={newStore} />;
    const wrapper = mount(component);
    const socialLink = wrapper.find(SocialLinks);

    expect(socialLink.find('.alert-danger').exists()).toBe(true);
  });
});
