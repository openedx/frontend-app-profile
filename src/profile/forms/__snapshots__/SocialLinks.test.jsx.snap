// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialLinks /> calls social links with edit mode bio 1`] = `
<div
  className="pgn-transition-replace-group position-relative mb-5"
  style={
    Object {
      "height": null,
    }
  }
>
  <div
    style={
      Object {
        "padding": ".1px 0",
      }
    }
  >
    <div
      aria-labelledby="social-links-label"
      role="dialog"
    >
      <form
        aria-labelledby="editing-form"
        onSubmit={[Function]}
      >
        <div
          className="editable-item-header mb-2"
        >
          <h2
            className="edit-section-header"
            id="social-links-label"
          >
            Social Links
          </h2>
        </div>
        <div
          id="social-error-feedback"
        />
        <ul
          className="list-unstyled"
        >
          <li
            className="form-group"
          >
            <label
              htmlFor="social-facebook"
            >
              Facebook
            </label>
            <input
              aria-describedby="social-error-feedback"
              className="form-control"
              id="social-facebook"
              name="facebook"
              onChange={[Function]}
              type="text"
              value="https://www.facebook.com/aloha"
            />
          </li>
          <li
            className="form-group"
          >
            <label
              htmlFor="social-twitter"
            >
              Twitter
            </label>
            <input
              aria-describedby="social-error-feedback"
              className="form-control"
              id="social-twitter"
              name="twitter"
              onChange={[Function]}
              type="text"
              value="https://www.twitter.com/ALOHA"
            />
          </li>
        </ul>
        <div
          className="d-flex flex-row-reverse flex-wrap justify-content-end align-items-center"
        >
          <div
            className="form-group d-flex flex-wrap"
          >
            <label
              className="col-form-label"
              htmlFor="visibilitySocialLinks"
            >
              Who can see this:
            </label>
            <span
              className="d-flex align-items-center"
            >
              <span
                className="d-inline-block ml-1 mr-2"
                style={
                  Object {
                    "width": "1.5rem",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-eye-slash fa-w-20 "
                  data-icon="eye-slash"
                  data-prefix="far"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 640 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M634 471L36 3.51A16 16 0 0 0 13.51 6l-10 12.49A16 16 0 0 0 6 41l598 467.49a16 16 0 0 0 22.49-2.49l10-12.49A16 16 0 0 0 634 471zM296.79 146.47l134.79 105.38C429.36 191.91 380.48 144 320 144a112.26 112.26 0 0 0-23.21 2.47zm46.42 219.07L208.42 260.16C210.65 320.09 259.53 368 320 368a113 113 0 0 0 23.21-2.46zM320 112c98.65 0 189.09 55 237.93 144a285.53 285.53 0 0 1-44 60.2l37.74 29.5a333.7 333.7 0 0 0 52.9-75.11 32.35 32.35 0 0 0 0-29.19C550.29 135.59 442.93 64 320 64c-36.7 0-71.71 7-104.63 18.81l46.41 36.29c18.94-4.3 38.34-7.1 58.22-7.1zm0 288c-98.65 0-189.08-55-237.93-144a285.47 285.47 0 0 1 44.05-60.19l-37.74-29.5a333.6 333.6 0 0 0-52.89 75.1 32.35 32.35 0 0 0 0 29.19C89.72 376.41 197.08 448 320 448c36.7 0 71.71-7.05 104.63-18.81l-46.41-36.28C359.28 397.2 339.89 400 320 400z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </span>
              <select
                className="d-inline-block w-auto form-control"
                id="visibilitySocialLinks"
                name="visibilitySocialLinks"
                onChange={[Function]}
                type="select"
                value="private"
              >
                <option
                  value="private"
                >
                  Just me
                </option>
                <option
                  value="all_users"
                >
                  Everyone on localhost
                </option>
              </select>
            </span>
          </div>
          <div
            className="form-group flex-shrink-0 flex-grow-1"
          >
            <button
              aria-disabled={false}
              aria-live="assertive"
              className="pgn__stateful-btn pgn__stateful-btn-state-pending btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="submit"
            >
              <span
                className="d-flex align-items-center justify-content-center"
              >
                <span
                  className="pgn__stateful-btn-icon"
                >
                  <span
                    className="pgn__icon icon-spin"
                  >
                    <svg
                      aria-hidden={true}
                      fill="none"
                      focusable={false}
                      height={24}
                      role="img"
                      viewBox="0 0 24 24"
                      width={24}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M22 12A10 10 0 116.122 3.91l1.176 1.618A8 8 0 1020 12h2z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  Saving
                </span>
              </span>
            </button>
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`<SocialLinks /> calls social links with edit mode certificates 1`] = `
<div
  className="pgn-transition-replace-group position-relative mb-5"
  style={
    Object {
      "height": null,
    }
  }
>
  <div
    style={
      Object {
        "padding": ".1px 0",
      }
    }
  >
    <div
      className="editable-item-header mb-2"
    >
      <h2
        className="edit-section-header"
        id={null}
      >
        Social Links
        <button
          className="float-right px-0 btn btn-link btn-sm"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "marginTop": "-.35rem",
            }
          }
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-pencil-alt fa-w-16 mr-1"
            data-icon="pencil-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Edit
        </button>
      </h2>
      <p
        className="mb-0"
      >
        <span
          className="ml-auto small text-muted"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-eye-slash fa-w-20 "
            data-icon="eye-slash"
            data-prefix="far"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M634 471L36 3.51A16 16 0 0 0 13.51 6l-10 12.49A16 16 0 0 0 6 41l598 467.49a16 16 0 0 0 22.49-2.49l10-12.49A16 16 0 0 0 634 471zM296.79 146.47l134.79 105.38C429.36 191.91 380.48 144 320 144a112.26 112.26 0 0 0-23.21 2.47zm46.42 219.07L208.42 260.16C210.65 320.09 259.53 368 320 368a113 113 0 0 0 23.21-2.46zM320 112c98.65 0 189.09 55 237.93 144a285.53 285.53 0 0 1-44 60.2l37.74 29.5a333.7 333.7 0 0 0 52.9-75.11 32.35 32.35 0 0 0 0-29.19C550.29 135.59 442.93 64 320 64c-36.7 0-71.71 7-104.63 18.81l46.41 36.29c18.94-4.3 38.34-7.1 58.22-7.1zm0 288c-98.65 0-189.08-55-237.93-144a285.47 285.47 0 0 1 44.05-60.19l-37.74-29.5a333.6 333.6 0 0 0-52.89 75.1 32.35 32.35 0 0 0 0 29.19C89.72 376.41 197.08 448 320 448c36.7 0 71.71-7.05 104.63-18.81l-46.41-36.28C359.28 397.2 339.89 400 320 400z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          Just me
        </span>
      </p>
    </div>
    <ul
      className="list-unstyled"
    >
      <li
        className="form-group"
      >
        <a
          className="font-weight-bold"
          href="https://www.facebook.com/aloha"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-facebook fa-w-16 mr-2"
            data-icon="facebook"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Facebook
        </a>
      </li>
      <li
        className="form-group"
      >
        <a
          className="font-weight-bold"
          href="https://www.twitter.com/ALOHA"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-twitter fa-w-16 mr-2"
            data-icon="twitter"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Twitter
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<SocialLinks /> calls social links with edit mode goals 1`] = `
<div
  className="pgn-transition-replace-group position-relative mb-5"
  style={
    Object {
      "height": null,
    }
  }
>
  <div
    style={
      Object {
        "padding": ".1px 0",
      }
    }
  >
    <div
      className="editable-item-header mb-2"
    >
      <h2
        className="edit-section-header"
        id={null}
      >
        Social Links
      </h2>
    </div>
    <ul
      className="list-unstyled"
    >
      <li
        className="mb-4"
      >
        <div>
          <button
            className="pl-0 text-left btn btn-link"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-plus fa-w-14 fa-xs mr-2"
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Add Facebook
          </button>
        </div>
      </li>
      <li
        className="mb-4"
      >
        <div>
          <button
            className="pl-0 text-left btn btn-link"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-plus fa-w-14 fa-xs mr-2"
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Add Twitter
          </button>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<SocialLinks /> calls social links with edit mode socialLinks 1`] = `
<div
  className="pgn-transition-replace-group position-relative mb-5"
  style={
    Object {
      "height": null,
    }
  }
>
  <div
    style={
      Object {
        "padding": ".1px 0",
      }
    }
  >
    <div
      className="editable-item-header mb-2"
    >
      <h2
        className="edit-section-header"
        id={null}
      >
        Social Links
        <button
          className="float-right px-0 btn btn-link btn-sm"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "marginTop": "-.35rem",
            }
          }
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-pencil-alt fa-w-16 mr-1"
            data-icon="pencil-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Edit
        </button>
      </h2>
      <p
        className="mb-0"
      >
        <span
          className="ml-auto small text-muted"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-eye-slash fa-w-20 "
            data-icon="eye-slash"
            data-prefix="far"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M634 471L36 3.51A16 16 0 0 0 13.51 6l-10 12.49A16 16 0 0 0 6 41l598 467.49a16 16 0 0 0 22.49-2.49l10-12.49A16 16 0 0 0 634 471zM296.79 146.47l134.79 105.38C429.36 191.91 380.48 144 320 144a112.26 112.26 0 0 0-23.21 2.47zm46.42 219.07L208.42 260.16C210.65 320.09 259.53 368 320 368a113 113 0 0 0 23.21-2.46zM320 112c98.65 0 189.09 55 237.93 144a285.53 285.53 0 0 1-44 60.2l37.74 29.5a333.7 333.7 0 0 0 52.9-75.11 32.35 32.35 0 0 0 0-29.19C550.29 135.59 442.93 64 320 64c-36.7 0-71.71 7-104.63 18.81l46.41 36.29c18.94-4.3 38.34-7.1 58.22-7.1zm0 288c-98.65 0-189.08-55-237.93-144a285.47 285.47 0 0 1 44.05-60.19l-37.74-29.5a333.6 333.6 0 0 0-52.89 75.1 32.35 32.35 0 0 0 0 29.19C89.72 376.41 197.08 448 320 448c36.7 0 71.71-7.05 104.63-18.81l-46.41-36.28C359.28 397.2 339.89 400 320 400z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          Just me
        </span>
      </p>
    </div>
    <ul
      className="list-unstyled"
    >
      <li
        className="form-group"
      >
        <a
          className="font-weight-bold"
          href="https://www.facebook.com/aloha"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-facebook fa-w-16 mr-2"
            data-icon="facebook"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Facebook
        </a>
      </li>
      <li
        className="form-group"
      >
        <a
          className="font-weight-bold"
          href="https://www.twitter.com/ALOHA"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-twitter fa-w-16 mr-2"
            data-icon="twitter"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Twitter
        </a>
      </li>
    </ul>
  </div>
</div>
`;
